traversal/GraphTraversal.java

```diff
@@ -221,30 +220,30 @@ public void predicate(Identifier.Variable attribute, TypeQLToken.Predicate token
         }
 
         public void predicate(Identifier.Variable attribute, TypeQLToken.Predicate.Equality token, Boolean value) {
-            Predicate.Value.Numerical predicate = Predicate.Value.Numerical.of(token, PredicateArgument.Value.BOOLEAN);
+            Predicate.Value.Numerical<Boolean> predicate = Predicate.Value.Numerical.of(token, PredicateArgument.Value.BOOLEAN);
             parameters.pushValue(attribute, predicate, new Parameters.Value(value));
             structure.thingVertex(attribute).props().predicate(predicate);
         }
 
         public void predicate(Identifier.Variable attribute, TypeQLToken.Predicate.Equality token, Long value) {
-            Predicate.Value.Numerical predicate = Predicate.Value.Numerical.of(token, PredicateArgument.Value.LONG);
+            Predicate.Value.Numerical<Long> predicate = Predicate.Value.Numerical.of(token, PredicateArgument.Value.LONG);
             parameters.pushValue(attribute, predicate, new Parameters.Value(value));
             structure.thingVertex(attribute).props().predicate(predicate);
         }
 
         public void predicate(Identifier.Variable attribute, TypeQLToken.Predicate.Equality token, Double value) {
             long longValue = Math.round(value);
-            if (Predicate.compareDoubles(value, longValue) == 0) {
+            if (Encoding.ValueType.DOUBLE.comparator().compare(value, (double) longValue) == 0) {
```

Comment:
this is a sneaky equality check to see if the long is close enough to a long to be considered a long